src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,216 +1,84 @@
 // (C) 2020 GoodData Corporation
 import * as React from "react";
 import without = require("lodash/without");
+import isEmpty = require("lodash/isEmpty");
+import get = require("lodash/get");
 import * as classNames from "classnames";
 import { Execution } from "@gooddata/typings";
+import { stringToFloat } from "../../../helpers/utils";
 import { IntlWrapper } from "../../core/base/IntlWrapper";
 import { IntlTranslationsProvider, ITranslationsComponentProps } from "../../core/base/TranslationsProvider";
-import { stringToFloat } from "../../../helpers/utils";
 import PushpinSizeLegend from "./legends/PushpinSizeLegend";
 import { getGeoDataIndex } from "../../../helpers/geoChart";
 import { IGeoConfig, IGeoDataIndex } from "../../../interfaces/GeoChart";
-import { COLOR, SIZE, SEGMENT_BY } from "../../../constants/bucketNames";
+import { SIZE } from "../../../constants/bucketNames";
 import { TOP } from "../../visualizations/chart/legend/PositionTypes";
-import Legend, { ILegendProps } from "../../visualizations/chart/legend/Legend";
 
-export interface IChartLegendProps {
+export interface IGeoChartLegendRendererProps {
     config: IGeoConfig;
     execution: Execution.IExecutionResponses;
     locale: string;
     position?: string;
 }
-export default class GeoChartLegendRenderer extends React.PureComponent<IChartLegendProps> {
-    constructor(props: IChartLegendProps) {
-        super(props);
-    }
-
+export default class GeoChartLegendRenderer extends React.PureComponent<IGeoChartLegendRendererProps> {
```

Comment:
I believe no state, for now, means we could use functional/stateless component, like
```
function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): React.ReactElement {...}
```